server/src/main/java/org/elasticsearch/common/io/stream/DelayableWriteable.java

```diff
@@ -46,7 +47,7 @@
      * when {@link #expand()} is called.
      */
     public static <T extends Writeable> DelayableWriteable<T> delayed(Writeable.Reader<T> reader, StreamInput in) throws IOException {
-        return new Serialized<>(reader, in.getVersion(), in.namedWriteableRegistry(), in.readBytesReference());
+        return new Serialized<>(reader, in.getVersion(), in.namedWriteableRegistry(), in.readReleasableBytesReference());
```

Comment:
This would literally create O(100M) sized byte arrays in a recent user issue, completely locking up GC on the coordinating node.